### PR TITLE
feat(clk): add simvars for Clock CHR and ET

### LINKED ISSUE #6049: Add simulation variables for the Clock
### Aircraft Version

Stable

### Description

Currently the Clock/Chronograph has two events: A32NX_CHRONO_TOGGLE and A32NX_CHRONO_RST.
The display is done via JS or something?
It would be nice to have all the events/vars at our disposal as explained on the Documentation: https://docs.flybywiresim.com/pilots-corner/a32nx-briefing/flight-deck/front/clock/

### References (optional)

_No response_

### Additional info (optional)

_No response_

### Discord Username (optional)

Wokkel#7194

### PR DESCRIPTION
**Pull request by ssewell**
Fixes #6049

## Summary of Changes
Adds simvars for the CHR and ET displays in the Clock module, using a value of -1 when no content is displayed.
 * A32NX_CHRONO_ELAPSED_TIME
 * A32NX_CHRONO_ET_ELAPSED_TIME

Naming was chosen with consistency in mind based on the existing simvars and events in Chrono.tsx and ElapsedTime.tsx.

## Screenshots (if necessary)
No visual changes as only new simvars were added.

## References
N/A

## Additional context
Changes were made to include simvars along with the existing functionality without changing the existing state implementation. Using the simvars exclusively in lieu of the React State hooks was considered but introduced visual judder between two consecutive values at times. 

These new simvars enable cockpit builders to keep their Clock hardware displays in sync with FBW's representation.

Discord username (if different from GitHub): AntennaRex#9906

## Testing instructions

1. Ensure the existing CHR and ET features on the Clock continue to behave normally.
2. Enable MSFS 2020 DevMode to inspect the two new simvars (A32NX_CHRONO_ELAPSED_TIME and A32NX_CHRONO_ET_ELAPSED_TIME) to see the seconds elapsed for the Clock's CHR and ET displays, respectively. Each simvar should be set to -1 when the corresponding display on the Clock is blank.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: docs/a320-simvars.md [hunk 1]
```diff
@@ -1102,6 +1102,21 @@
...
+      0 or greater | Seconds elapsed
+      -1 | Empty value
+
+- A32NX_CHRONO_ET_ELAPSED_TIME
+    - Number
+    - Clock instrument ET display time elapsed
+      Value | Meaning
+      --- | ---
+      0 or greater | Seconds elapsed
+      -1 | Empty value
```
Comment (Benjozork) [inline docs/a320-simvars.md:1119]:
```suggestion
      -1 | Empty value

```

File: src/instruments/src/Clock/Components/Chrono.tsx [hunk 2]
```diff
@@ -13,11 +13,14 @@ export const Chrono = () => {
     const [prevTime, setPrevTime] = useState(absTime);
 
     const [elapsedTime, setElapsedTime] = useState<null | number>(null);
+    const [elapsedSimvarTime, setElapsedSimvarTime] = useSimVar('L:A32NX_CHRONO_ELAPSED_TIME', 'number');
```
Comment (Benjozork) [inline src/instruments/src/Clock/Components/Chrono.tsx:16]: Does not seem like we use the value itself

```suggestion
    const [, setElapsedSimvarTime] = useSimVar('L:A32NX_CHRONO_ELAPSED_TIME', 'number');
```

File: src/instruments/src/Clock/Components/ElapsedTime.tsx [hunk 3]
```diff
@@ -9,6 +9,7 @@ export const ElapsedTime = () => {
     const [ltsTest] = useSimVar('L:A32NX_OVHD_INTLT_ANN', 'bool', 250);
     const [dcEssIsPowered] = useSimVar('L:A32NX_ELEC_DC_ESS_BUS_IS_POWERED', 'bool', 250);
     const [elapsedKnobPos] = useInteractionSimVar('L:A32NX_CHRONO_ET_SWITCH_POS', 'number', 'A32NX_CHRONO_ET_POS_CHANGED');
+    const [elapsedSimvarTime, setElapsedSimvarTime] = useSimVar('L:A32NX_CHRONO_ET_ELAPSED_TIME', 'number');
```
Comment (Benjozork) [inline src/instruments/src/Clock/Components/ElapsedTime.tsx:12]: Ditto

```suggestion
    const [, setElapsedSimvarTime] = useSimVar('L:A32NX_CHRONO_ET_ELAPSED_TIME', 'number');
```

**Review by Benjozork — CHANGES_REQUESTED** [review]

**Review by Benjozork — APPROVED** [review]

**Comment by mico975** [comment]
Quality Assurance Tester Report

Discord          : mico#3145
Object of testing: #6790
Tier of Testing  : 1
Date             : 23/02/2022

Testing Process: 
Ensure the existing CHR and ET features on the Clock continue to behave normally. - OK
Enable MSFS 2020 DevMode to inspect the two new simvars (A32NX_CHRONO_ELAPSED_TIME and A32NX_CHRONO_ET_ELAPSED_TIME) to see the seconds elapsed for the Clock's CHR and ET displays, respectively. Each simvar should be set to -1 when the corresponding display on the Clock is blank. - OK

Negatives:       
On initial load the CHR and ET vars are set to 0 (displays are blank, so it's still ok), after reset they are set to -1.

Testing Results: 
Passed

Conclusions: 
Works

**Comment by frankkopp** [comment]
@ssewell please make sure to open an issue on the docs repo to update the API documentation 
Best to link to to your PR for details. 
Thanks

**Comment by ssewell** [comment]
@frankkopp Done!

**Comment by ssewell** [comment]
> Negatives:
> On initial load the CHR and ET vars are set to 0 (displays are blank, so it's still ok), after reset they are set to -1.

@mico975 Good catch on this! I should initialize these to -1 before they're used.

**Comment by owen2007** [comment]
Quality Assurance Trainee Report

Discord          : (owen_#4883)
Object of testing: (#PR 6790)
Tier of Testing  : (1)
Date             : (1/03/2022)

Testing Process: 
(Detail is needed)

Negatives:       
There where no negatives that i found when conducting the test.

Testing Results: 
The test passed 

Conclusions: 
In concluion the simvars both simvars where set to -1

**Pull request merged**